tests/Unit/Support/Traits/AssertCssTest.php

```diff
@@ -0,0 +1,289 @@
+     */
+    public function getCssNeedleRegExpInsertsOptionalWhitespace($contentToInsertAround, $otherContent)
+    {
+        $result = TestingAssertCss::getCssNeedleRegExp(
+            $otherContent . $contentToInsertAround . $otherContent
+        );
+
+        $quotedOtherContent = preg_quote($otherContent, '/');
+
+        static::assertSame(
+            '/' . $quotedOtherContent . '\\s*+' . preg_quote($contentToInsertAround, '/') . '\\s*+'
+                . $quotedOtherContent . '/',
+            $result
+        );
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function needleFoundDataProvider()
+    {
+        $cssStrings = [
+            'unminified CSS' => 'body { color: green; }',
+            'minified CSS' => 'body{color: green;}',
+            'CSS with extra spaces' => 'body  {  color: green;  }',
+            'CSS with linefeeds' => "body\n{\ncolor: green;\n}",
+            'CSS with Windows line endings' => "body\r\n{\r\ncolor: green;\r\n}",
+        ];
+
+        $datasets = [];
+        foreach ($cssStrings as $needleDescription => $needle) {
+            foreach ($cssStrings as $haystackDescription => $haystack) {
+                $datasets[$needleDescription . ' in ' . $haystackDescription] = [$needle, $haystack];
+            }
+        }
+        return $datasets;
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function needleNotFoundDataProvider()
+    {
+        return [
+            'CSS part with "{" not in CSS' => ['p {', 'body { color: green; }'],
+            'CSS part with "}" not in CSS' => ['color: red; }', 'body { color: green; }'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @dataProvider needleFoundDataProvider
+     */
+    public function assertContainsCssPassesTestIfNeedleFound($needle, $haystack)
+    {
+        TestingAssertCss::assertContainsCss($needle, $haystack);
+    }
+
+    /**
+     * @test
+     *
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @dataProvider needleNotFoundDataProvider
+     *
+     * @expectedException PHPUnit_Framework_ExpectationFailedException
+     */
+    public function assertContainsCssFailsTestIfNeedleNotFound($needle, $haystack)
+    {
+        TestingAssertCss::assertContainsCss($needle, $haystack);
+    }
+
+    /**
+     * @test
+     *
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @dataProvider needleNotFoundDataProvider
+     */
+    public function assertNotContainsCssPassesTestIfNeedleNotFound($needle, $haystack)
+    {
+        TestingAssertCss::assertNotContainsCss($needle, $haystack);
+    }
+
+    /**
+     * @test
+     *
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @dataProvider needleFoundDataProvider
+     *
+     * @expectedException PHPUnit_Framework_ExpectationFailedException
```

Comment:
Please add the missing leading Backslash to the expection class name to make it obvious to the reader that this is not namespaced.

Reply:
Ditto for the other `@expectedException`.

Reply:
When I originally tried with the leading `\` it didn't work.  But now it does.

Reply:
It's magic! ;-)